test/opt/add_var_for_func_call_param_test.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gmock/gmock.h"
+#include "test/opt/pass_fixture.h"
+#include "test/opt/pass_utils.h"
+
+namespace spvtools {
+namespace opt {
+namespace {
+
+using AddVarFuncCallParamTest = PassTest<::testing::Test>;
```

Comment:
Please use the Match tests for future tests: https://github.com/KhronosGroup/SPIRV-Tools/blob/2b2b0282af25779a5ae5dfc08a1c4e3be5a8ae17/test/opt/scalar_replacement_test.cpp#L989

There are a lot of legacy tests that do what you have here, but we prefer new tests to use the Matching because it calls out the specific changes, and do not have to be updated if the id numbers change by 1.